commit 273dc279306c2b69fbc1407cc695397e4b015152
Author: Will Newton <will.newton@linaro.org>
Date:   Fri May 16 15:22:58 2014 +0100

    ld/aarch64: xfail non-PIC shared object tests on aarch64
    
    aarch64 does not support building non-PIC shared objects, so mark
    tests for these as xfail.
    
    ld/testsuite/ChangeLog:
    
    2014-05-20  Will Newton  <will.newton@linaro.org>
    
    	* ld-shared/shared.exp: Mark non-PIC shared object tests
    	as xfail on aarch64.

#-- a/ld/testsuite/ChangeLog
#++ b/ld/testsuite/ChangeLog
#@ -1,3 +1,8 @@
+2014-05-20  Will Newton  <will.newton@linaro.org>
+
+	* ld-shared/shared.exp: Mark non-PIC shared object tests
+	as xfail on aarch64.
+
 2014-05-19  Andreas Tobler  <andreast@fgznet.ch>
 
 	* ld-elf/shared.exp: Introduce the extralibs variable to control
diff --git a/ld/testsuite/ld-shared/shared.exp b/ld/testsuite/ld-shared/shared.exp
index 9996ece..c723ef7 100644
--- a/ld/testsuite/ld-shared/shared.exp
+++ b/ld/testsuite/ld-shared/shared.exp
@@ -244,6 +244,7 @@ if ![ld_compile "$CC $CFLAGS $SHCFLAG" $srcdir/$subdir/main.c $tmpdir/mainnp.o]
 	if [ string match $shared_needs_pic "yes" ] {
 	    setup_xfail "arm*-*-linux*"
 	}
+	setup_xfail "aarch64*-*-linux*"
 	shared_test shnp "shared (non PIC)" mainnp.o sh1np.o sh2np.o shared
 
 	# Test ELF shared library relocations with a non-zero load
@@ -270,6 +271,7 @@ if ![ld_compile "$CC $CFLAGS $SHCFLAG" $srcdir/$subdir/main.c $tmpdir/mainnp.o]
 	if [ string match $shared_needs_pic "yes" ] {
 	    setup_xfail "arm*-*-linux*"
 	}
+	setup_xfail "aarch64*-*-linux*"
 	shared_test shnp "shared (non PIC, load offset)" \
 		mainnp.o sh1np.o sh2np.o shared \
 		"-T $srcdir/$subdir/elf-offset.ld"
@@ -326,6 +328,7 @@ if ![ld_compile "$CC $CFLAGS $SHCFLAG $picflag" $srcdir/$subdir/main.c $tmpdir/m
 	    if [ string match $shared_needs_pic "yes" ] {
 		setup_xfail "arm*-*-linux*"
 	    }
+	    setup_xfail "aarch64*-*-linux*"
 	    shared_test shmpnp "shared (PIC main, non PIC so)" mainp.o sh1np.o sh2np.o shared
 	}
     } else {
